sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs

```diff
@@ -3,20 +3,61 @@
 {
     /// <summary>Represents an identifier in Azure Communication Services.</summary>
-#pragma warning disable CS0659 // Type overrides Object.Equals(object o) but does not override Object.GetHashCode()
     public abstract class CommunicationIdentifier : IEquatable<CommunicationIdentifier>
-#pragma warning restore CS0659 // Type overrides Object.Equals(object o) but does not override Object.GetHashCode()
     {
+        /// <summary>
+        /// Returns the canonical string representation of the <see cref="CommunicationIdentifier"/>.
+        /// You can use the <see cref="RawId"/> for encoding the identifier and then use it as a key in a database.
+        /// </summary>
+        public virtual string RawId { get; }
+
+        /// <inheritdoc />
+        public override int GetHashCode() => RawId.GetHashCode();
+
         /// <inheritdoc />
         public abstract bool Equals(CommunicationIdentifier other);
 
         /// <inheritdoc />
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override bool Equals(object obj)
             => obj is CommunicationIdentifier other && Equals(other);
+
+        /// <summary>
+        /// Creates a <see cref="CommunicationIdentifier"/> from a given rawId.
+        /// When storing rawIds, use this function to restore the identifier that was encoded in the rawId.
+        /// </summary>
+        /// <param name="rawId">The rawId to be translated to its identifier representation.</param>
+        /// <returns>Returns <see cref="CommunicationUserIdentifier"/>, <see cref="PhoneNumberIdentifier"/>, <see cref="MicrosoftTeamsUserIdentifier"/>, or <see cref="UnknownIdentifier"/> based on the identifier type.</returns>
+        public static CommunicationIdentifier FromRawId(string rawId)
+        {
+            Argument.AssertNotNullOrEmpty(rawId, nameof(rawId));
+
+            if (rawId.StartsWith("4:", StringComparison.OrdinalIgnoreCase))
+            {
+                return new PhoneNumberIdentifier($"+{rawId.Substring("4:".Length)}");
+            }
+
+            var segments = rawId.Split(':');
+            if (segments.Length < 3)
+                return new UnknownIdentifier(rawId);
+
+            var prefix = $"{segments[0]}:{segments[1]}:";
+            var suffix = rawId.Substring(prefix.Length);
+
+            return prefix switch
+            {
+                "8:teamsvisitor:" => new MicrosoftTeamsUserIdentifier(suffix, true),
+                "8:orgid:" => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Public),
+                "8:dod:" => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Dod),
+                "8:gcch:" => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Gcch),
+                "8:acs:" or "8:spool:" or "8:dod-acs:" or "8:gcch-acs:" => new CommunicationUserIdentifier(rawId),
+                _ => new UnknownIdentifier(rawId),
+            };
+        }
```

Comment:
I think we can create helper class, the future changes to this will help not to update this CommunicationIdentifier.

Reply:
Why don't you like updating the `CommunicationIdentifier` class? 
I'd prefer to keeping it this way instead of creating a helper class. Currently, it's nicely aligned with having the serialization logic in the specific implementations of the `CommunicationIdentifier.RawId`.

Reply:
Sounds good @petrsvihlik